checks/prgenv/mpi.py

```diff
@@ -53,16 +31,23 @@ class MpiInitTest(rfm.RegressionTest):
     )
     prebuild_cmds += ['module list']
     time_limit = '2m'
-    maintainers = ['JG', 'AJ']
+    maintainers = ['@jgphpc']
     tags = {'production', 'craype'}
 
     @run_after('init')
     def set_cpp_flags(self):
         self.build_system.cppflags = self.cppflags[self.required_thread]
 
+    @run_after('setup')
+    def skip_if_no_mpi(self):
+        self.skip_if('mpi' not in self.current_partition.features,
+                     'skip partition with no mpi')
+        self.skip_if('mpi' not in self.current_environ.features,
+                     'skip environ with no mpi')
+
```

Comment:
We can remove this and just add it in the `valid_prog_environs`, `valid_systems`.